src/utils/can-cards-connect/tests/index.test.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  PassageCard,
+  DeadendCard,
+  TunnelCard,
+} from "../../../models/cards/path-cards";
+import { Sides } from "../../../models/cards/card";
+import canCardsConnect, { getOppositeSide } from "..";
+
+const sideName = ["top", "right", "bottom", "left"];
+
+describe("check two cards can connect", () => {
+  describe("when connecting with an empty space", () => {
+    it("can connect", () => {
+      const card = new PassageCard([
+        Sides.top,
+        Sides.right,
+        Sides.bottom,
+        Sides.left,
+      ]);
+      const adjacentCard = undefined;
+      expect(canCardsConnect(0, adjacentCard, card)).toBe(true);
+    });
+  });
+
+  const commonTestCases = [
+    [
+      "which always has common connectors",
+      [Sides.top, Sides.right, Sides.bottom, Sides.left],
+      [Sides.top, Sides.right, Sides.bottom, Sides.left],
+      [true, true, true, true],
+    ],
+    [
+      "which never has common connectors",
+      [Sides.right, Sides.left],
+      [Sides.top, Sides.bottom],
+      [false, false, false, false],
+    ],
+    [
+      "which always has common connectors or common spaces",
+      [Sides.top, Sides.bottom],
+      [Sides.top, Sides.bottom],
+      [true, true, true, true],
+    ],
+  ] as [string, Sides[], Sides[], boolean[]][];
+
+  describe.each([
```

Comment:
Using `describe.each` a bunch in this file to confirm a the combination of:
 - `PassageCard`s and `DeadendCard`s work the same
 - Cards with connectors on corresponding sides will connect
 - Cards without connectors on corresponding sides will connect
 - Cards with a mismatch of connectors and not connectors will not connect